{"content":"---\ntitle: \"'Finally' cannot appear outside a 'Try' statement\"\nms.date: 07/20/2015\nf1_keywords: \n  - \"vbc30382\"\n  - \"bc30382\"\nhelpviewer_keywords: \n  - \"BC30382\"\nms.assetid: 0314d8d2-18bc-4bbd-858c-0a18408b52fd\n---\n# 'Finally' cannot appear outside a 'Try' statement\n`Finally` is used to complete a `Try...Catch...Finally` block; hence it can only appear once at the end of the block. Either you have an unnecessary `Finally`, or your `Finally` statement appears outside the bounds of its corresponding `Try` block.  \n  \n **Error ID:** BC30382  \n  \n## To correct this error  \n  \n1.  Locate and remove the unnecessary `Finally` statement.  \n  \n2.  Move the `Finally` statement to the appropriate location in your code.  \n  \n## See also\n\n- [Try...Catch...Finally Statement](../../visual-basic/language-reference/statements/try-catch-finally-statement.md)\n","nodes":[{"pos":[4,210],"embed":true,"restype":"x-metadata","content":"title: \"'Finally' cannot appear outside a 'Try' statement\"\nms.date: 07/20/2015\nf1_keywords: \n  - \"vbc30382\"\n  - \"bc30382\"\nhelpviewer_keywords: \n  - \"BC30382\"\nms.assetid: 0314d8d2-18bc-4bbd-858c-0a18408b52fd","nodes":[{"content":"'Finally' cannot appear outside a 'Try' statement","nodes":[{"pos":[0,49],"content":"'Finally' cannot appear outside a 'Try' statement","nodes":[{"content":"'Finally' cannot appear outside a 'Try' statement","pos":[0,49]}]}],"path":["title"],"nosxs":false}],"yml":true},{"pos":[217,266],"content":"'Finally' cannot appear outside a 'Try' statement","linkify":"'Finally' cannot appear outside a 'Try' statement","nodes":[{"content":"'Finally' cannot appear outside a 'Try' statement","pos":[0,49]}]},{"content":"<ph id=\"ph1\">`Finally`</ph> is used to complete a <ph id=\"ph2\">`Try...Catch...Finally`</ph> block; hence it can only appear once at the end of the block.","pos":[267,384],"source":"`Finally` is used to complete a `Try...Catch...Finally` block; hence it can only appear once at the end of the block."},{"content":"Either you have an unnecessary <ph id=\"ph1\">`Finally`</ph>, or your <ph id=\"ph2\">`Finally`</ph> statement appears outside the bounds of its corresponding <ph id=\"ph3\">`Try`</ph> block.","pos":[385,515],"source":" Either you have an unnecessary `Finally`, or your `Finally` statement appears outside the bounds of its corresponding `Try` block."},{"pos":[522,543],"content":"<bpt id=\"p1\">**</bpt>Error ID:<ept id=\"p1\">**</ept> BC30382","source":"**Error ID:** BC30382"},{"pos":[552,573],"content":"To correct this error","linkify":"To correct this error","nodes":[{"content":"To correct this error","pos":[0,21]}]},{"pos":[583,637],"content":"Locate and remove the unnecessary <ph id=\"ph1\">`Finally`</ph> statement.","source":"Locate and remove the unnecessary `Finally` statement."},{"pos":[647,717],"content":"Move the <ph id=\"ph1\">`Finally`</ph> statement to the appropriate location in your code.","source":"Move the `Finally` statement to the appropriate location in your code."},{"pos":[726,734],"content":"See also","linkify":"See also","nodes":[{"content":"See also","pos":[0,8]}]},{"pos":[738,852],"content":"<bpt id=\"p1\">[</bpt>Try...Catch...Finally Statement<ept id=\"p1\">](../../visual-basic/language-reference/statements/try-catch-finally-statement.md)</ept>","source":"[Try...Catch...Finally Statement](../../visual-basic/language-reference/statements/try-catch-finally-statement.md)"}]}